pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -572,13 +575,34 @@ func shouldH2Upgrade(clusterName string, direction model.TrafficDirection, port
 }
 
 // setH2Options make the cluster an h2 cluster by setting http2ProtocolOptions.
-// TODO(https://github.com/istio/istio/issues/29735) remove nolint
-// nolint: staticcheck
 func setH2Options(cluster *cluster.Cluster) {
-	if cluster == nil || cluster.Http2ProtocolOptions != nil {
+	if cluster == nil || cluster.GetTypedExtensionProtocolOptions() != nil {
 		return
 	}
-	cluster.Http2ProtocolOptions = &core.Http2ProtocolOptions{
+	cluster.TypedExtensionProtocolOptions = httpProtocolOptions()
+}
+
+func httpProtocolOptions() map[string]*any.Any {
+	options, err := ptypes.MarshalAny(
+		&http.HttpProtocolOptions{
+			UpstreamProtocolOptions: &http.HttpProtocolOptions_ExplicitHttpConfig_{
+				ExplicitHttpConfig: &http.HttpProtocolOptions_ExplicitHttpConfig{
+					ProtocolConfig: &http.HttpProtocolOptions_ExplicitHttpConfig_Http2ProtocolOptions{
+						Http2ProtocolOptions: http2ProtocolOptions(),
+					},
+				},
+			},
+		})
+	if err != nil {
+		return nil
+	}
+	return map[string]*any.Any{
```

Comment:
weird, I have never seen this structure before. Why would they need a map, doesn't any already include the type?

Can you use options.TypeUrl as the key or does it include the googleapis prefix?

Reply:
Map looks weird and strictly not needed here. https://github.com/envoyproxy/envoy/blob/main/api/envoy/config/cluster/v3/cluster.proto#L805 - they have plans to move it to list.

Reply:
> Can you use options.TypeUrl as the key or does it include the googleapis prefix?

options does not have a typeUrl. For now I have added it to v3 types.